### PR TITLE
Major package upgrade

### PR DESCRIPTION
**Pull request by anizeani**
All polkadot & encointer packages uptodate 15.01.2022 (encointer v0.6.0)
explorer bump to 0.5.0
React v17 
Code adapted to work with new rpc's and types in encointer v0.6.0
Massive cleanup in packages and lint errors/warnings, since lint didn't work for a while (dependency issues)
couldn't test sidebar with phone! 

and all other packages are updated to the latest versions, besides react-leaftlet, because this would require to replace another component (LMap), and  semantic-ui-css, which is kind of on the newest change, but if upgraded, it will get a breaking change (double semicolon that webpack can't compile) that hasn't been fixed yet (discussed and solved here: https://stackoverflow.com/questions/70367443/create-react-app-with-typescript-failing-to-compile-after-importing-semantic-ui)
However, lint --fix still doesn't fix all files, had to manually fix them (useSubstrate i.e.)
should also fix that in another issue. 

And: DEP_WEBPACK_DEV_SERVER_ON_AFTER_SETUP_MIDDLEWARE i didn't find yet where the mentioned option is set
and the long base64 string I didn't figure out yet, why this is output (no error, just for some reason outputted).

Maybe a separate cleanup issue would be necessary.

### CODE REVIEW

File: src/Map.js
Comment (clangenb) [inline src/Map.js:320]: I would rather call it `fmtCid` than `CidComplete`.

File: src/Map.js
Comment (clangenb) [inline src/Map.js:416]: This is good; leave it here. It will be helpful. In general, I prefer to use function strings, which is less verbose than using `+` to concatenate strings, e.g.:

```
console.log(`state participants are: ${JSON.stringify(state.participants))}`, [state]);
```

Why do you use `[state]` separately after the comma? `console.log` has the args: `path`, `msg`. Hence you would use it like:

`console.log('function/class where the logging takes place', 'log msg')`

File: src/map/MapSidebar.js
Comment (clangenb) [inline src/map/MapSidebar.js:71]: Please change the variable name `_` to match the returned value: `balanceEntry`

**Review by clangenb — COMMENTED** [review]
This is only a code review. As far as I can tell, this looks good. I will test now with a local node.

**Comment by clangenb** [comment]
I did some tests, and some things are not working as they should. I run `bootstrap_demo_community.py`, which does:

* register 3 participants (bootstrappers: alice, bob, charlie)
* performs 1 meetup
* goes to registering phase and claims issuance
* then I used the cli.py helper script in the node: `./cli.py register-alice-bob-charlie`, and then go to the assigning phase: this is what I see:

image

It fails to display the values from the last ceremony, I'd expect `participants registered in last ceremony: 3` and `meetups assigned in last ceremony: 1`, but both values are 0. 

Note: There have been some registry purging issues in the node [PR](https://github.com/encointer/pallets/pull/111), which should, however, not be related to this. Just to be sure, I will update the encointer-pallets in the node, and test again.

**Comment by clangenb** [comment]
Ok, I updated the pallets in the node, and it did solve the issue with the last ceremony's state. I did observe the expected:  `participants registered in last ceremony: 3` and `meetups assigned in last ceremony: 1`.

Interestingly, I am 99% sure that I observed the following deprecation warning the first time I started the server too, but I did not see it the following times.

```
[DEP_WEBPACK_DEV_SERVER_ON_AFTER_SETUP_MIDDLEWARE] DeprecationWarning: 'onAfterSetupMiddleware' option
is deprecated. Please use the 'setupMiddlewares' option.
```

Conclusively, I think this PR can be merged after the issues from the code-review are solved. I would like to have the linter issues being fixed after merging this one, as it should be a minor issue.

File: package.json
Comment (anizeani) [reply package.json:50]: This was a recommendation in the terminal, also it was suggested in several threads. Eslint dependent is also set in the yarn.lock..

File: src/Map.js
Comment (anizeani) [reply src/Map.js:282]: Agree

File: src/Map.js
Comment (anizeani) [reply src/Map.js:93]: Ah, i actually wanted to add some descriptions, so it would be clearer for someone else (and for me if i have to get back to it. I'll try to phrase it clearer)

File: src/Map.js
Comment (anizeani) [reply src/Map.js:416]: I'm not sure you know how react works here:
We are using states here (not regular variables), which are initialized with useState(state,setterOfState) =... 
You can not simply log states in react after you set them, since setting them occurs in a different time during rendering. 
The only way i found to securely log them, is with using the useEffect function. 
The function looks like this:
UseEffect( () => {},[stateOfChoise1, stateOfChoise2.. ] )
Every time at least one of the states defined in the array change, it will trigger the useEffect to excecute
UseEffect

Comment (clangenb) [reply src/Map.js:416]: Ah sorry, I misread the brackets, and after your explanation it makes sense to me. Thanks!

**Comment by anizeani** [comment]
> 

I'm not sure, why sometimes the deprecation warnings are not displayed anymore in the terminal. Are you using webstorm? For me, it sometimes also stopped, but I do not think its a change in the packages, but rather a setting to not show it everytime.. 
About the linting, it shouldnt be a big issue, but i am not sure, if we want the warning of double quotes vs single quotes, and I also didnt manage to autofix it and i would like to have it autofixed ;)

**Comment by clangenb** [comment]
It suprises me that this can't be autofixed via the commandline. But it should work if you do it via the formatting of the IDE.

**Review by clangenb — COMMENTED** [review]
Nice improvement again. Two last questions/comments, then we are good to go.

File: src/map/MapNodeSwitchWidget.js
Comment (anizeani) [reply src/map/MapNodeSwitchWidget.js:212]: I haven't checked, this is a function. I'm not sure whether it will be needed or not, and out of laziness/low priority didn't cjeck if its affecting anything, so i just removed the warning

File: src/Map.js
Comment (anizeani) [reply src/Map.js:278]: The reason why I added this comparison is because the .toNumber() function failed everytime we were in phase 0. The reason is, that the other phases (meetup/attestation) return numbers, but assignements return {bootrappers: number, endorsees: number, newbies: number}..
See the difference?

Comment (clangenb) [reply src/Map.js:278]: Ah ok, i did not really get that piece of code. Now I understand. 👍

File: src/map/MapNodeSwitchWidget.js
Comment (anizeani) [reply src/map/MapNodeSwitchWidget.js:212]: removed it

File: src/substrate-lib/useSubstrate.js
Comment (anizeani) [reply src/substrate-lib/useSubstrate.js:28]: so basically remove the whole types decleration, can/should i also remove the ...options?

File: src/Map.js
Comment (anizeani) [reply src/Map.js:278]: not so hirsch afterall...

**Review by clangenb — APPROVED** [review]
Looks good to me now.

**Review by brenzi — CHANGES_REQUESTED** [review]
tested against fresh reset of Gesell running on 0.8.0

I get 
`API/INIT: Error: FATAL: Unable to initialize the API: createType(Metadata):: decodeU8a: failed at 0x0ef902000c1c7370… on metadata:: Unable to create Enum via index 14, in V0, V1, V2, V3, V4, V5, V6, V7, V8, V9, V10, V11, V12, V13`

**Comment by brenzi** [comment]
fails for me:
```
Uncaught Error: Module build failed (from ./node_modules/babel-loader/lib/index.js):
Error: [BABEL] /mnt/d/encointer/explorer/node_modules/@pmmmwh/react-refresh-webpack-plugin/client/ReactRefreshEntry.js: Cannot find module 'levenary'
Require stack:
- /mnt/d/encointer/explorer/node_modules/babel-preset-react-app/node_modules/@babel/helper-compilation-targets/lib/index.js
- /mnt/d/encointer/explorer/node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/debug.js
- /mnt/d/encointer/explorer/node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/index.js
- /mnt/d/encointer/explorer/node_modules/babel-preset-react-app/dependencies.js
- /mnt/d/encointer/explorer/node_modules/@babel/core/lib/config/files/module-types.js
- /mnt/d/encointer/explorer/node_modules/@babel/core/lib/config/files/configuration.js
- /mnt/d/encointer/explorer/node_modules/@babel/core/lib/config/files/index.js
- /mnt/d/encointer/explorer/node_modules/@babel/core/lib/index.js
- /mnt/d/encointer/explorer/node_modules/@rollup/plugin-babel/dist/index.js
- /mnt/d/encointer/explorer/node_modules/workbox-build/build/lib/bundle.js
- /mnt/d/encointer/explorer/node_modules/workbox-webpack-plugin/build/generate-sw.js
- /mnt/d/encointer/explorer/node_modules/workbox-webpack-plugin/build/index.js
- /mnt/d/encointer/explorer/node_modules/react-scripts/config/webpack.config.js
- /mnt/d/encointer/explorer/node_modules/react-scripts/scripts/start.js (While processing: "/mnt/d/encointer/explorer/node_modules/babel-preset-react-app/dependencies.js")
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:101:18)
    at Object.<anonymous> (/mnt/d/encointer/explorer/node_modules/babel-preset-react-app/node_modules/@babel/helper-compilation-targets/lib/index.js:41:40)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    node_modules bundle.js:40
    factory react refresh:6
    Webpack 3
```

**Comment by clangenb** [comment]
Works for me, I can't reproduce @brenzi's error. Did you run yarn install?

Verified that:
*  Successfully launch an instance with `yarn start` and can display ceremony data.
* Veried that the binary builds successfully with `yarn build`

Hence, looks good to me.

**Comment by clangenb** [comment]
It is very 

> fails for me on WSL ubuntu 20.04 & firefox & node 14:
> 
>     factory react refresh:6
>     Webpack 3

Webpack 3 should not be a part of the project, this is very weird.

I have WSL ubuntu 20.04 & firefox & node **16.13.1**:

**Comment by anizeani** [comment]
I can't reproduce the error. 
works for me on 
WSL Ubuntu 18.04 & Chrome & node 16.10.0, yarn 1.22.10
I tested with yarn install, yarn build & yarn start. 
should all work

**Comment by anizeani** [comment]
> fails for me on WSL ubuntu 20.04 & firefox & node 14:
> 
> ```
> Uncaught Error: Module build failed (from ./node_modules/babel-loader/lib/index.js):
> Error: [BABEL] /mnt/d/encointer/explorer/node_modules/@pmmmwh/react-refresh-webpack-plugin/client/ReactRefreshEntry.js: Cannot find module 'levenary'
> Require stack:
> - /mnt/d/encointer/explorer/node_modules/babel-preset-react-app/node_modules/@babel/helper-compilation-targets/lib/index.js
> - /mnt/d/encointer/explorer/node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/debug.js
> - /mnt/d/encointer/explorer/node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/index.js
> - /mnt/d/encointer/explorer/node_modules/babel-preset-react-app/dependencies.js
> - /mnt/d/encointer/explorer/node_modules/@babel/core/lib/config/files/module-types.js
> - /mnt/d/encointer/explorer/node_modules/@babel/core/lib/config/files/configuration.js
> - /mnt/d/encointer/explorer/node_modules/@babel/core/lib/config/files/index.js
> - /mnt/d/encointer/explorer/node_modules/@babel/core/lib/index.js
> - /mnt/d/encointer/explorer/node_modules/@rollup/plugin-babel/dist/index.js
> - /mnt/d/encointer/explorer/node_modules/workbox-build/build/lib/bundle.js
> - /mnt/d/encointer/explorer/node_modules/workbox-webpack-plugin/build/generate-sw.js
> - /mnt/d/encointer/explorer/node_modules/workbox-webpack-plugin/build/index.js
> - /mnt/d/encointer/explorer/node_modules/react-scripts/config/webpack.config.js
> - /mnt/d/encointer/explorer/node_modules/react-scripts/scripts/start.js (While processing: "/mnt/d/encointer/explorer/node_modules/babel-preset-react-app/dependencies.js")
>     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
>     at Function.Module._load (internal/modules/cjs/loader.js:746:27)
>     at Module.require (internal/modules/cjs/loader.js:974:19)
>     at require (internal/modules/cjs/helpers.js:101:18)
>     at Object.<anonymous> (/mnt/d/encointer/explorer/node_modules/babel-preset-react-app/node_modules/@babel/helper-compilation-targets/lib/index.js:41:40)
>     at Module._compile (internal/modules/cjs/loader.js:1085:14)
>     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
>     at Module.load (internal/modules/cjs/loader.js:950:32)
>     at Function.Module._load (internal/modules/cjs/loader.js:790:12)
>     at Module.require (internal/modules/cjs/loader.js:974:19)
>     node_modules bundle.js:40
>     factory react refresh:6
>     Webpack 3
> ```
did you freshly install the dependencies and build it? 
yarn install --> yarn start/build

**Pull request merged**
